packages/design-system-react/package.json

```diff
@@ -87,8 +87,8 @@
     "@metamask/utils": "^11.4.2",
     "@solana/addresses": "^2.0.0",
     "bitcoin-address-validation": ">=2.0.0",
-    "react": "^16.0.0",
-    "react-dom": "^16.0.0",
+    "react": "^17.0.0 || ^18.0.0",
+    "react-dom": "^17.0.0 || ^18.0.0",
```

Comment:
Here we are aligning react versions with out consuming projects
- [ds workspace v18](https://github.com/MetaMask/metamask-design-system/blob/main/apps/storybook-react/package.json#L38-L39)
- [extension v17](https://github.com/consensys-vertical-apps/metamask-portfolio/blob/main/package.json#L71)
- [portfolio v18](https://github.com/consensys-vertical-apps/metamask-portfolio/blob/main/package.json#L71)
- [developer dashboard v18](https://github.com/MetaMask/metamask-developer-dashboard/blob/main/package.json#L118)